elastic/kibana
# [Security Solution][Notes] - add create note to flyout notes tab

## Summary

This PR continues to add functionality to the new notes feature for Security Solution. It focuses on adding a add note section to the expandable flyout Notes tab.

The new `AddNote` component within the Notes tab does the following:
- displays the `MarkdownEditor` shared component from Security Solution to allow users to create notes with OsQuery, timeline links and more
- an add button. The add button will be in loading state while the creation of the note is happening. At the same time, the `NotesList` component renders a loading spinner between the last note and the `AddNote` component, where the newly created note will appear once the call finishes.

https://github.com/elastic/kibana/assets/17276605/3a1f4e0e-63cd-4970-b95f-52492abe4071

### How to test

- make sure the feature flag is enabled in your `kibana.yml` file: `xpack.securitySolution.enableExperimental: ['notesEnabled']`
- open the expandable flyout for an alert, click on the Expand details button to expand the left section then navigate to the new Notes tab

_Note: as the server side code isn't implemented yet for fetching alerts by document, the created note are persisted but cannot be retrieved, so refreshing the page makes it look like they are not._

https://github.com/elastic/security-team/issues/9375
https://github.com/elastic/security-team/issues/9605

### Checklist
